src/numerics/include/metaphysicl/numberarray.h

```diff
@@ -184,6 +184,12 @@ class NumberArray
     return returnval;
   }
 
+  void zero()
```

Comment:
Allowing *manual* zeroing is a good idea, though, of course.  Curious question: is there a reason to prefer the manual loop over the ```std::fill``` idiom in the constructor above?